File: app/src/main/java/com/example/tasos/icalendare/database/ICalendarDatabase.java
package com.example.tasos.icalendare.database;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.migration.Migration;
import android.content.Context;

@Database(entities = {Contact.class,TypeOfEvent.class}, version = 1)
public abstract class ICalendarDatabase extends RoomDatabase {

    public abstract ContactDao contactDao();
    public abstract TypeOfEventDao typeOfEventDao();

    public static ICalendarDatabase INSTANCE;

    static ICalendarDatabase getDatabase(final Context context){
        if (INSTANCE == null) {
            synchronized (ICalendarDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            ICalendarDatabase.class, "icalendar_database")
                            .fallbackToDestructiveMigration()
                            .build();

                }
            }
        }
        return INSTANCE;
    }
/*
    static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(SupportSQLiteDatabase database) {
            // Since we didn't alter the table, there's nothing else to do here.
        }
    };
    */
}

File: app/src/main/java/com/example/tasos/icalendare/Database/ICalendarDatabase.kt
package com.example.tasos.icalendare.Database

import android.arch.persistence.room.*
import android.content.Context

@Database(entities = arrayOf(Contact::class, TypeOfEvent::class, Events::class), version = 23)
@TypeConverters(CalendarTypeConverter::class)
abstract class ICalendarDatabase : RoomDatabase() {

    abstract fun contactDao(): ContactDao

    abstract fun typeOfEventDao(): TypeOfEventDao

    abstract fun eventsDao(): EventsDao

    companion object {

        private var INSTANCE: ICalendarDatabase? = null

        @Synchronized
        fun getInstance(context: Context): ICalendarDatabase {
            if (INSTANCE == null) {
                synchronized(ICalendarDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                ICalendarDatabase::class.java, "icalendar_database")
                                .fallbackToDestructiveMigration()
                                .allowMainThreadQueries()
                                .build()

                    }
                }
            }
            return INSTANCE!!
        }

        fun destroyInstance() {
            INSTANCE = null
        }
    }
    /*
    static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(SupportSQLiteDatabase database) {
            // Since we didn't alter the table, there's nothing else to do here.
        }
    };
    */


}
